label_studio_converter/converter.py

```diff
@@ -408,11 +411,20 @@ def convert_to_csv(self, input_data, output_dir, is_dir=True, **kwargs):
                 record['agreement'] = item['agreement']
             records.append(record)
 
-        df = pd.DataFrame.from_records(records)
-        if records:
-            # convert annotation_id with Null values as Int, refer to https://pandas.pydata.org/pandas-docs/stable/user_guide/gotchas.html#support-for-integer-na
-            df['annotation_id'] = df['annotation_id'].astype(pd.Int64Dtype())
-        df.to_csv(output_file, index=False, **kwargs)
+        # Previously we were using pandas dataframe to_csv() but that produced incorrect JSON so writing manually
```

Comment:
Thanks for catching the TSV use case! I have handled that now. 

For escaping quotes, that was not happening at all previously. For fields that are being json serialized, it will happen automatically. For fields that aren't, I don't think they are expected to be ingested into tools that will need quotes to be escaped. We can make another ticket about this to make sure in the future

Reply:
Code looks good to me now! Is there a test we can write for the JSON encoding functionality?